### ISSUE TITLE
Update destroyed all my pages

### ISSUE THREAD

**Issue by bdorr1105** [docmost/docmost#262]
Not sure what happened, but updating to 0.3.0 absolutely destroyed all the hard work I did documenting things. All the pages I have are there, however, no data is there at all. This is absolutely infuriating. All of this is down the drain

![Screenshot 2024-09-05 075815](https://github.com/user-attachments/assets/1e3ba91f-7fa0-4cb3-ba5a-4dc4663c1301)

**Comment by Philipinho** [comment]
The collaboration websocket is not connecting. Please upgrade to v0.3.1 which fixes this.

**Comment by bdorr1105** [comment]
I am on version 0.3.1

**Comment by Philipinho** [comment]
Your data is not gone. The editor retrieves the page's content from the collaboration websocket. If it fails to connect, the editor returns a blank page. I should make this more obvious on connection failure.

Please open a page and send me a screenshot of the "WS" console tab which shows the "/collab" path.

**Comment by bdorr1105** [comment]
Does this work for what you needed to see

![Screenshot 2024-09-05 080849](https://github.com/user-attachments/assets/80fb6e0e-5457-4592-a56a-85fddb55a372)

**Comment by bdorr1105** [comment]
This is the WS tab more specifically

![Screenshot 2024-09-05 081058](https://github.com/user-attachments/assets/a55cf713-4104-4290-854c-023acd100e1a)

**Comment by bdorr1105** [comment]
![Screenshot 2024-09-05 081155](https://github.com/user-attachments/assets/c0525605-1399-4e87-a27b-152238dd2d35)

**Comment by Philipinho** [comment]
Is your reverse proxy setup to work with WebSockets?

**Comment by bdorr1105** [comment]
Yes, I am using nginx proxy manager, everything was fine prior to the 0.3.0 update.

![Screenshot 2024-09-05 081847](https://github.com/user-attachments/assets/5279cbe6-6572-4690-abcc-1a200e17bb23)

**Comment by Philipinho** [comment]
Have you tried a different browser? from the screenshots, the WebSocket URL is fine.

**Comment by bdorr1105** [comment]
yes, I have tried edge with a fresh login, all of my data is gone or does not exist. I do not know why only my subpages exist, but a lot of the data was added when I was using 0.3.0

**Comment by bdorr1105** [comment]
so far what I see is inconsistency, everything seemes to be cached in the browser session but not actually saving. I went incognito and all the changes it shows in the regular session are def not being saved at all

**Comment by Philipinho** [comment]
The real-time collaboration has offline support. When the websocket fails to connect, the changes are stored in the browser's Indexeddb. It syncs locally in real-time across tabs too. 
Upon establishing a successful WebSocket connection, the local state is merged with the server state.

The issue here is the websocket failing to connect. I am not sure why yet.
